Sources/AppStoreConnectCLI/Services/Operations/ListDevicesOperation.swift

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+import AppStoreConnect_Swift_SDK
+import Combine
+
+struct ListDevicesOperation: APIOperation {
+
+    struct Options {
+       let filterName: [String]
+       let filterPlatform: [Platform]
+       let filterUDID: [String]
+       let filterStatus: DeviceStatus?
+       let sort: Devices.Sort?
+       let limit: Int?
+    }
+
+    private let options: Options
+
+    init(options: Options) {
+        self.options = options
+    }
+
+    func execute(with requestor: EndpointRequestor) throws -> AnyPublisher<[Device], Error> {
+
+        var filters = [Devices.Filter]()
+
+        if !options.filterName.isEmpty {
+            filters.append(.name(options.filterName))
+        }
+
+        if !options.filterPlatform.isEmpty {
+            // API Device attributes use the BundleIdPlatform enum,
+            // rather than a Platform, so there is no support for
+            // tvOs or watchOs.
+            // This appears to be an API issue.
+            filters.append(.platform(options.filterPlatform))
+        }
+
+        if !options.filterUDID.isEmpty {
+            filters.append(.udid(options.filterUDID))
+        }
+
+        if let filterStatus = options.filterStatus {
+            filters.append(.status([filterStatus]))
+        }
+
+        let sort = [options.sort].compactMap { $0 }
+
+        guard let limit = options.limit else {
+            return requestor.requestAllPages {
+                    .listDevices(
+                        filter: filters,
+                        sort: sort,
+                        next: $0
+                    )
+                }
+                .map { $0.flatMap { $0.data } }
+                .eraseToAnyPublisher()
+        }
+
+        return requestor.request(
+                .listDevices(
```

Comment:
```suggestion
               .listDevices(
```

Reply:
I guess think this intent might be correct, I use auto intent then give it a tab then it will look like this 🤔

Reply:
It just looked a bit weird to me :-) .listDevices aligned with requestor looks better to me..but upto you..